Common/src/toolboxes/signal_processing_toolbox.cpp

```diff
@@ -27,11 +27,115 @@
 
 #include "../../include/toolboxes/signal_processing_toolbox.hpp"
 
+//WindowingTools
+/*! \brief Returns the value of a windowing function given by fctIdx at CurTimeIdx with given endTimeIdx (i.e. endTimeIdx=nTimeIter, if one starts  windowing at time t =0.) */
+su2double WindowingTools::GetWndWeight(int fctIdx, unsigned long CurTimeIdx, unsigned long endTimeIdx){
+    switch (fctIdx){
+      case 1: return HannWindow(CurTimeIdx, endTimeIdx);
+      case 2: return HannSquaredWindow(CurTimeIdx, endTimeIdx);
+      case 3: return BumpWindow(CurTimeIdx, endTimeIdx);
+      default:return 1.0;
+    }
+}
```

Comment:
Here the type of `fctIdx` should be the enumerator, and the "cases" should be actual names not the numbers.
All these functions can be const or static if you use the globally defined pi.

Reply:
GetWndWeight now uses the enum WINDOW_FUNCTION.